lib/inertia_rails/renderer.rb

```diff
@@ -77,6 +82,15 @@ def deep_transform_values(hash, proc)
       hash.transform_values {|value| deep_transform_values(value, proc)}
     end
 
+    def deferred_props_keys
+      return if rendering_partial_component?
+
+      @props.select { |_, prop| prop.is_a?(DeferProp) }
```

Comment:
while I really love the functional approach, personally, I would optimize this with smth like:
```ruby
  @props.each_with_object({}) do |(key, prop), result|
    (result[prop.group] ||= []) << key.to_s if prop.is_a?(DeferProp)
  end
```

Reply:
Yes much better